src/main/scala/com/vngrs/etl/Loader.scala

```diff
@@ -0,0 +1,31 @@
+package com.vngrs.etl
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+
+trait Loader[A] {
+
+  protected[etl] def apply(input: RDD[A]): Loader[A]
+
+}
+
+object Loader {
+  def apply[A](f: RDD[A] => Unit): Loader[A] = new Loader[A]() {
+    override def apply(data: RDD[A]): Loader[A] = {
+      f(data)
+      this
+    }
+  }
+
+  def file(path: String): Loader[String] = new Loader[String] {
```

Comment:
Factory methods on companion objects will provide an entry point to the functionality we provide officially. Implementing them via anonymous classes or named classes is the topic of another discussion. Users still can define custom functionality with inheriting the desired trait.